openassessment/xblock/openassessmentblock.py

```diff
@@ -404,6 +404,19 @@ def should_use_user_state(self, upload_urls):
             and self.user_state_upload_data_enabled() \
             and self.file_upload_response
 
+    def should_get_all_files_urls(self, upload_urls):
+        """
+        Returns a boolean to decide if all the file submitted by a learner in a block should be obtained.
+
+        Following conditions should be true for boolean to be true:
+        1. The waffle flag/switch is enabled
+        2. the file upload is required or optional
+        3. the file data from submission is missing information
+        """
+        return not any(upload_urls) \
```

Comment:
what about?
`return not any(upload_urls) and all([self.all_files_urls_enabled(),self.file_upload_response])`

Reply:
`all()` checks every case, even if it encounters a Falsy value. The current arrangements will return early with the counter of a Falsy value

Reply:
`all_files_urls_enabled` should be renamed to indicate that it is checking a waffle flag/switch. It will make the code more readable.